neo/rawio/axonarawio.py

```diff
@@ -0,0 +1,374 @@
+
+    def get_channel_offset(self, chan_id):
+        """ 
+        In the .bin file, channels are arranged in a strange order. This method takes
+        a channel index as input and returns the actual offset for the channel in the
+        memory map (self._raw_signals).
+        """
+        channnel_memory_offset = np.array(
+            [32, 33, 34, 35, 36, 37, 38, 39, 0, 1, 2, 3, 4, 5,
+             6, 7, 40, 41, 42, 43, 44, 45, 46, 47, 8, 9, 10, 11,
+             12, 13, 14, 15, 48, 49, 50, 51, 52, 53, 54, 55, 16, 17,
+             18, 19, 20, 21, 22, 23, 56, 57, 58, 59, 60, 61, 62, 63,
+             24, 25, 26, 27, 28, 29, 30, 31]
+        )
+        return channnel_memory_offset[chan_id - 1]
+
+    def read_datetime(self):
+        """ 
+        Creates datetime object (y, m, d, h, m, s) from .set file header 
+        """
+        with open(self.set_file, 'r', encoding=self.set_file_encoding) as f:
+            for line in f:
+                if line.startswith('trial_date'):
+                    date_string = re.findall(r'\d+\s\w+\s\d{4}$', line)[0]
+                if line.startswith('trial_time'):
+                    time_string = line[len('trial_time')+1::].replace('\n', '')
+
+        return datetime.datetime.strptime(date_string + ', ' + time_string, \
+            "%d %b %Y, %H:%M:%S")
+
+    def get_channel_gain(self):
+        """ 
+        Read gain for each channel from .set file and return in list of integers 
+
+        This is actually not the gain_ch value from the .set file, but the conversion
+        factor from raw data to uV.
+
+        Formula for .eeg and .X files, presumably also .bin files:    
+        
+        1000*adc_fullscale_mv / (gain_ch*128)
+        """
+        gain_list = []
+
+        with open(self.set_file, encoding='cp1252') as f:
+            for line in f:
+                if line.startswith('ADC_fullscale_mv'):
+                    adc_fullscale_mv = int(line.split(" ")[1])
+                if line.startswith('gain_ch'):
+                    gain_list.append(np.float32(re.findall(r'\d*', line.split(' ')[1])[0]))
+        
+        return [1000*adc_fullscale_mv/(gain*128) for gain in gain_list]
+
+    def get_signal_chan_header(self):
+        """
+        Returns a 1 dimensional np.array of tuples with one entry for each channel
+        that recorded data. Each tuple contains the following information:
+
+        channel name (1a, 1b, 1c, 1d, 2a, 2b, ...; with num=tetrode, letter=electrode), 
+        channel id (1, 2, 3, 4, 5, ... N), 
+        sampling rate, 
+        data type (int16), 
+        unit (uV), 
+        gain,
+        offset, 
+        stream id
+        """
+        active_tetrode_set = self.get_active_tetrode()
+        num_active_tetrode = len(active_tetrode_set)
+
+        elec_per_tetrode = 4
+        letters = ['a', 'b', 'c', 'd']
+        dtype = 'int16'
+        units = 'uV'
+        gain_list = self.get_channel_gain()
+        offset = 0  # What is the offset? 
```

Comment:
The offset is when you have uint16 for instance and you need to do
`sig_scaled = sig_raw * gain + offset`